lib/plugins/entry-files-manifest.js

```diff
@@ -9,26 +9,42 @@
 
 'use strict';
 
-const EntryFilesManifestPlugin = require('../webpack/entry-files-manifest-plugin');
 const PluginPriorities = require('./plugin-priorities');
-const path = require('path');
 const sharedEntryTmpName = require('../utils/sharedEntryTmpName');
-const manifestKeyPrefixHelper = require('../utils/manifest-key-prefix-helper');
+const AssetsPlugin = require('assets-webpack-plugin');
+
+function cleanFilePaths(assets) {
+    const tmpAssets = assets;
```

Comment:
this does not create a copy of the data structure. It only create a separate reference to the same object. So AFAICT, this is useless here, and you could as well use `assets` all the time (the other solution is that it is broken)

Reply:
Maybe something like this instead? (I didn't test it so there may be some errors)

```js
const processedAssets = assets
    .filter((asset, name) => name !== sharedEntryTmpName);
    .map(asset => {
        return asset.map(typeAssets => {
            if (!Array.isArray(typeAssets )) {
                typeAssets = [typeAssets];
            }

            return typeAssets.map(buildPath => buildPath.replace(/^\//g, '');
        });
    });

return JSON.stringify(processedAssets);
```

It would also put everything in an array like the previous code, even if there is only one file.

Reply:
I tried to do the fancy snippet you had but it came out with filter is not a function. assets is a json object when it comes in. since I am doing dynamic keys that is why it looks like I am using an array when it is a json object. I checked the object prototype and I dont see any really helpful methods. I can try implementing them my self in some helper module. I dont know how fancy we want to go with it.

Reply:
Yeah, nevermind, you'd have to filter assets.keys() first and then rebuild the object... not sure that's worth it if directly mutating `assets` isn't an issue.

Reply:
I think mutating it is fine if things work. But then, we should not create a `tmpAssets` variable, which make people reading the code *think* we have 2 separate data structures (while we only have a second name for the same one)